unidist/core/backends/mpi/core/communication.py

```diff
@@ -179,6 +207,33 @@ def recv_operation_type(comm):
 # ---------------------------------
 
 
+def _send_complex_data_impl(comm, s_data, raw_buffers, len_buffers, dest_rank):
+    """
+    Send already serialized complex data.
+
+    Parameters
+    ----------
+    comm : object
+        MPI communicator object.
+    s_data : object
+        Serialized data as bytearray.
+    raw_buffers : list
+        Pickle buffers list, out-of-band data collected with pickle 5 protocol.
+    len_buffers : list
+        Size of each buffer from `raw_buffers` list.
+    dest_rank : int
+        Target MPI process to transfer data.
+    """
+    # Send message pack bytestring
+    mpi_send_buffer(comm, len(s_data), s_data, dest_rank)
+    # Send the necessary metadata
+    mpi_send_object(comm, len(raw_buffers), dest_rank)
+    for raw_buffer in raw_buffers:
+        mpi_send_buffer(comm, len(raw_buffer.raw()), raw_buffer, dest_rank)
+    # TODO: do not send if raw_buffers is zero
+    mpi_send_object(comm, len_buffers, dest_rank)
```

Comment:
There is at least 4 sends. Can we lessen these?

Reply:
*At most* 4 sends, but usually 3 sends, that can be reduced to 2 send - added a TODO for this task. Already mentioned, that I tries to do it fast within this PR, but faced protocol errors and decided to make it separately.

Reply:
Create an issue please.

Reply:
#102 issue was created